modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt

```diff
@@ -2226,11 +2228,34 @@ open class PlaybackManager @Inject constructor(
                 if (isPlaying()) {
                     statsManager.addTotalListeningTime(UPDATE_TIMER_POLL_TIME)
                 }
+                if (playbackStateRelay.blockingFirst().isSleepTimerRunning && !sleepAfterEpisode) { // Does not apply to end of episode sleep time
+                    setupFadeOutWhenFinishingSleepTimer()
+                }
             }
             .switchMapCompletable { updateCurrentPositionRx() }
             .subscribeBy(onError = { Timber.e(it) })
     }
 
+    private fun setupFadeOutWhenFinishingSleepTimer() {
+        // Needs to run in main thread because of player getVolume
+        Handler(Looper.getMainLooper()).post {
+            val timeLeft = sleepTimer.timeLeftInSecs()
+            timeLeft?.let {
+                val fadeDuration = 5
+                val maxVolume = (player as? SimplePlayer)?.getVolume() ?: 1.0f
```

Comment:
I think this is the current volume and not max volume.

Reply:
@MiSikora I think I should rename this variable to `startVolume` because the current start volume is the one I start the fade out. What do you think?

Reply:
Yup, good idea! It makes even more sense since we use `1.0f` as a fallback.